### PR TITLE
Add Angular Accelerometer and Linear Velocity Sensor Interfaces to the Sensor Remapper

### PR DESCRIPTION
**Pull request by nicktrem**
- This PR will allow the angular acceleration and linear velocity measurements measured by a particular sensor to be extracted and used via the sensor remapper.

### CODE REVIEW

File: cmake/YarpIDL.cmake [hunk 1]
```diff
@@ -193,7 +193,7 @@ endfunction()
 function(YARP_IDL_TO_DIR)
 
   # Flag to control whether IDL generation is allowed.
-  option(ALLOW_IDL_GENERATION "Allow YARP to (re)build IDL files as needed" OFF)
+  option(ALLOW_IDL_GENERATION "Allow YARP to (re)build IDL files as needed" ON)
```
Comment (randaz81) [inline cmake/YarpIDL.cmake:196]: This value should not be modified

Comment (nicktrem) [reply cmake/YarpIDL.cmake]: Fixed in [3c5625e](https://github.com/robotology/yarp/pull/3149/commits/3c5625e2c8a430da81a5a96e593e305145f7ae02)

**Comment by randaz81** [comment]
Is this PR complete? @nicktrem @traversaro

File: src/devices/networkWrappers/multipleanalogsensorsserver/MultipleAnalogSensorsServer.cpp [hunk 2]
```diff
@@ -25,14 +25,16 @@ enum MAS_SensorTypeServer
...
+    ThreeAxisAngularAccelerometers=2,
+    ThreeAxisMagnetometers=3,
+    OrientationSensors=4,
+    TemperatureSensors=5,
+    SixAxisForceTorqueSensors=6,
+    ContactLoadCellArrays=7,
+    EncoderArrays=8,
+    SkinPatches=9,
+    PositionSensors=10,
+    LinearVelocitySensors=11
```
Comment (traversaro) [inline src/devices/networkWrappers/multipleanalogsensorsserver/MultipleAnalogSensorsServer.cpp:37]: Can we just add the new value at the end for avoiding changing the value of the constants unless strictly necessary?

**Comment by traversaro** [comment]
My only blocking comment is https://github.com/robotology/yarp/pull/3149/files#r1856392918 . Beside that, this introduces a communication incompatibility between YARP 3.10 and YARP master for `multipleanalogsensorsserver` and `multipleanalogsensorsclient`, but I guess this is kind of expected and documented, so once   https://github.com/robotology/yarp/pull/3149/files#r1856392918 is addressed this is ready for be reviewed for me.

File: src/devices/networkWrappers/multipleanalogsensorsserver/MultipleAnalogSensorsServer.cpp (hunk 2 above)
Comment (nicktrem) [reply src/devices/networkWrappers/multipleanalogsensorsserver/MultipleAnalogSensorsServer.cpp]: Fixed in [8ec2a36](https://github.com/robotology/yarp/pull/3149/commits/8ec2a36b969862e466e87ce3044598b332820df3)

**Comment by nicktrem** [comment]
The development of this branch has been completed. I have marked this PR as ready to review.

@randaz81 @traversaro please let me know if there are any additional requests, Thanks!

File: src/libYARP_dev/src/yarp/dev/MultipleAnalogSensorsInterfaces.h [hunk 3]
```diff
@@ -249,6 +289,50 @@ class YARP_dev_API yarp::dev::IPositionSensors
...
+     *
+     * @return false if an error occurred, true otherwise.
+     */
+    virtual bool getLinearVelocitySensorFrameName(size_t sens_index, std::string& frameName) const = 0;
+
+    /**
+     * Get the last reading of the linear velocity sensor as x y z.
+     *
+     * @param[in] sens_index The index of the specified sensor (should be between 0 and getNrOfLinearVelocitySensors()-1).
+     * @param[out] out The requested measure. The vector should be 3-dimensional. The measure is expressed in meters.
```
Comment (traversaro) [inline src/libYARP_dev/src/yarp/dev/MultipleAnalogSensorsInterfaces.h:325]:
```suggestion
     * @param[out] out The requested measure. The vector should be 3-dimensional. The measure is expressed in meters per second.
```

File: src/libYARP_dev/src/yarp/dev/MultipleAnalogSensorsInterfaces.h [hunk 4]
```diff
@@ -145,6 +147,44 @@ class YARP_dev_API yarp::dev::IThreeAxisLinearAccelerometers
...
+     * Get the name of the frame of the specified sensor.
+     * @return false if an error occurred, true otherwise.
+     */
+    virtual bool getThreeAxisAngularAccelerometerFrameName(size_t sens_index, std::string &frameName) const = 0;
+
+    /**
+     * Get the last reading of the specified sensor.
+     *
+     * @param[in] sens_index The index of the specified sensor (should be between 0 and getNrOfThreeAxisAngularAccelerometers()-1).
+     * @param[out] out The requested measure. The vector should be 3-dimensional. The measure is expressed in meters^2/seconds.
```
Comment (traversaro) [inline src/libYARP_dev/src/yarp/dev/MultipleAnalogSensorsInterfaces.h:179]:
```suggestion
     * @param[out] out The requested measure. The vector should be 3-dimensional. The measure is expressed in degrees^2/seconds.
```

File: src/libYARP_dev/src/yarp/dev/MultipleAnalogSensorsInterfaces.h [hunk 5]
```diff
@@ -145,6 +147,44 @@ class YARP_dev_API yarp::dev::IThreeAxisLinearAccelerometers
     virtual ~IThreeAxisLinearAccelerometers(){}
 };
 
+class YARP_dev_API yarp::dev::IThreeAxisAngularAccelerometers
+{
+public:
+    /**
+     * \brief Get the number of three axis linear accelerometers exposed by this device.
```
Comment (traversaro) [inline src/libYARP_dev/src/yarp/dev/MultipleAnalogSensorsInterfaces.h:154]:
```suggestion
     * \brief Get the number of three axis angular accelerometers exposed by this device.
```

**Review by traversaro — APPROVED** [review]

**Comment by traversaro** [comment]
@randaz81 for me the PR is good to go.

**Pull request merged**